src/Auth/Form/SignupForm.php

```diff
@@ -73,8 +73,12 @@ function (): Result {
                     $result->addError($this->translator->translate('validator.password.not.match'));
                 }
 
-                if ($result->getErrors() === [] && !$this->authService->signup($this->login, $this->password)) {
-                    $result->addError($this->translator->translate('validator.user.exist'));
+                if ($result->getErrors() === []) {
+                    try {
+                        $this->authService->signup($this->login, $this->password);
```

Comment:
Validation as creating a real user? Who made it? 😨

Reply:
I didn't like it either. But the changes would affect other files. This PR would be huge. So I left it as is.

Reply:
I agree that it's better to do separately. I was just confused by this code and couldn't get past.